crud/select/plan.lua

```diff
@@ -48,6 +49,46 @@ local function get_index_for_condition(space_indexes, space_format, condition)
     end
 end
 
+local function extract_sharding_key_from_conditions(conditions, ddl_sharding_key, space_indexes, space_field_positions)
```

Comment:
During discussion of implementation of #181 we decided to split sharding key for two terms:
- sharding key - is tuple fields used for bucket id calculation.
- sharding key definition - field names of tuple that used for sharding key. In DDL schema and in `box.space.ddl_sharding_key` user defines sharding key definition.

Seems in `extract_sharding_key_from_conditions()` you are passing sharding key defintion and I would rename second argument to sharding_key_def or something like that.

Reply:
It is what you call `sharding_key_as_index_obj` (it is a form of sharding key definition, sure) in PR #181, so I would name it in the same way. See https://github.com/tarantool/crud/pull/242#discussion_r756929428.

Reply:
Used `sharding_key_as_index_obj`